### PR TITLE
Issue 4849: filter for official document collection

### PR DESCRIPTION
**Pull request by Niicck**
https://github.com/cityofaustin/techstack/issues/4849

# Description

Alright. What have we done.

New Features:
- [ ] PaginationContainers now have an optional Date Filter. Yay. 
  - [ ] https://github.com/cityofaustin/janis/pull/873/files#diff-a6baaf58f2426fa64122109743d8587f984af6a9d28a2eab696124bd49ae6325
  - [ ] https://janis-v3-4849-filter.netlify.app/en/police-oversight/official-complaint-and-discipline-documents/
- [ ] We're using a library called useQueryParams in order to sync queryParams in our URL directly into our react state. No more edge case glitches or manually trying to keep them in sync. This makes it easy to have multiple components that rely on queryParam states and keep them all in sync.
  - [ ] This consequently allows pagination page and filter changes to be added to the browser history. So when you press the "Back" button, you'll go back to your previous filter settings.
  - [ ] This feature is used by the Pagination/Filter components, Search Page, and SearchBar.
  - [ ] Instead of using "#" to specify the start of a set of queryParams, we're following the standard "?" convention.
- [ ] Our PageSelector component on Pagination Pages is a bit simpler.
  - [ ] https://github.com/cityofaustin/janis/pull/873/files#diff-63baf2f63cb242a2b8d7554f7a36e3f178101aed413cfd517eead88603d5c1c8
  - [ ] This change was made possible by changing the structure of buildPageSelectorValues
    - [ ] https://github.com/cityofaustin/janis/pull/873/files#diff-965daf77109d9da9c3e91fc5f678ab7522bd474f83d5e0d0cc88af538e030cef
- [ ] Because we're using useQueryParams, some of the the PaginationContainer component logic got easier. It also applies the filter to the paginated results.
  - [ ] https://github.com/cityofaustin/janis/pull/873/files#diff-1ac7b91a8e1b1eb497cfab3030253faac74177974a32d4a0917ac217fc2b6727
  - [ ] The transition animations are a lot easier now. I ran this by Chase, he's cool with us just having an animation on FadeIn. That means it can be implemented purely in CSS. Hooray. 
    - [ ] https://github.com/cityofaustin/janis/pull/873/files#diff-4dc46bc71ac4d192ae4fc451be9f2d059cbee0b3007916f30c02a6a16f36025e
- [ ] We're using only 1 PaginationContainer for the entire site. No more copy/pasta. News, Search, Events, Official Docs all use the same one.
  - [ ] Got rid of the `fillTablet` and `smallMargins` props. Those can be done purely in CSS. Or not implemented in order to keep consistency for all PaginationContainers.
    - [ ] https://github.com/cityofaustin/janis/pull/873/files?file-filters%5B%5D=.js&file-filters%5B%5D=.json&file-filters%5B%5D=.lock&file-filters%5B%5D=.scss&file-filters%5B%5D=.sh&hide-deleted-files=true#diff-939dd8b486bde7c32f479969f4b98c84eb75bd15cecf727215de149d82b67489
- [ ] Created a `mobilePopupHelper`. This lets us reuse the logic we made for the Guide page in other components.
  - [ ] https://github.com/cityofaustin/janis/pull/873/files?file-filters%5B%5D=.js&file-filters%5B%5D=.json&file-filters%5B%5D=.lock&file-filters%5B%5D=.scss&file-filters%5B%5D=.sh&hide-deleted-files=true#diff-c0e799b2622ee6d940b91b214c5fa6dd11226532bc4540ce1908e1ea45a2537b
- SearchPage got easier to manage too because of useQueryParams.
  - [ ] https://github.com/cityofaustin/janis/pull/873/files?file-filters%5B%5D=.js&file-filters%5B%5D=.json&file-filters%5B%5D=.lock&file-filters%5B%5D=.scss&file-filters%5B%5D=.sh&hide-deleted-files=true#diff-f609ed88fbecdc37533a4e41e175aa6ad7098ce011ed116a1008e6e48f798c54
- [ ] scrollTransition can now scroll to the top of a specified element, not just the top of the page.
  - [ ] Also got rid of the opacity transition logic. That is now managed in CSS.
  - [ ] https://github.com/cityofaustin/janis/pull/873/files?file-filters%5B%5D=.js&file-filters%5B%5D=.json&file-filters%5B%5D=.lock&file-filters%5B%5D=.scss&file-filters%5B%5D=.sh&hide-deleted-files=true#diff-55ab85183cd1b61ca7e353198de30b30d65e110439a8a81d926664e94e5e81b3

Bonus things:
- [ ] Added eslint. Feel free to incorporate into your workflow/IDE.

# Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How can this be tested?
Check it out: 
  - [ ] Small collection: https://janis-v3-4849-filter.netlify.app/en/health-safety/policing-in-austin/police-feedback-and-records/nicks-small-collection/
  - [ ] Big collection: https://janis-v3-4849-filter.netlify.app/en/police-oversight/official-complaint-and-discipline-documents
# Checklist:
- [ ] Request reviewers
- [ ] Slack-out message for review
- [ ] Link this PR in the issue
- [ ] Moved card to *review* in zenhub

### CODE REVIEW

File: src/components/PageSections/Pagination/_Filter.scss [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+    These values will allow our calendar to resize proportionally
+    on any size mobile or tablet screen.
+    The width must be set on the parent container (.coa-filter__box).
+    This will enable our coa-filter__date-input-containers to resize with flex-grow
+    to match the width of our calendar.
+**/
+$coa-calendar-mobile-width: 75vw;
+$coa-calendar-mobile-height: calc(70vw - 20px);
+$coa-calendar-tablet-width: 50vw;
+$coa-calendar-tablet-height: calc(50vw - 20px);
```
Comment (Niicck) [inline src/components/PageSections/Pagination/_Filter.scss:141]: I was proud of this.

File: src/js/i18n/constants.js [hunk 2]
```diff
@@ -77,3 +79,53 @@ export const i18nalizeLinkTo = (to, langCode = DEFAULT_LANG) => {
...
+    "April",
+    "May",
+    "June",
+    "July",
+    "August",
+    "September",
+    "October",
+    "November",
+    "December",
+  ].map(month=>intl.formatMessage(i18n1[month]))
```
Comment (Niicck) [inline src/js/i18n/constants.js:100]: These are used in the Filter's DayPicker.

File: src/js/i18n/definitions.js [hunk 3]
```diff
@@ -578,3 +596,18 @@ export const news = defineMessages({
...
+  date: 'Date',
+  from: 'From',
+  to: 'To',
+  month: 'Month',
+  day: 'Day',
+  year: 'Year',
+  clearFilters: 'Clear filters',
+  applyFilters: 'Apply filters',
+  filteredResults: '{count} filtered results',
+  OneFilteredResult: '1 filtered result',
```
Comment (chiaberry) [inline src/js/i18n/definitions.js:612]: the nitpickiest thing I'll say is `oneFilteredResult`

File: src/js/helpers/date.js [hunk 4]
```diff
@@ -5,3 +5,19 @@ export const getDaysInOrder = () => {
...
+
+  So we convert a dateString from pagesArray "YYYY-MM-DD" into a valid Date.
+  new Date("YYYY-MM-DD") will not work consistently but
+  new Date("YYYY", "MM"-1, "DD") will work properly.
+**/
+export const createDateFromString = (dateString) => {
+  if (!dateString) return null;
+  let year, month, day;
+  [year, month, day] = dateString.split("-")
+  return new Date(year, month-1, day)
```
Comment (chiaberry) [inline src/js/helpers/date.js:22]: why is it month-1?

Comment (chiaberry) [reply src/js/helpers/date.js:22]: months are zero-indexed, but years and days are not

File: src/components/PageSections/Pagination/Filter.js [hunk 5]
```diff
@@ -0,0 +1,389 @@
...
+  }
+}
+
+const FilterMobilePopup = ({applyFilter, fromDate, toDate, lowerBound, upperBound}) => {
+  const intl = useIntl();
+  const [menuOpened, setMenuOpened] = useState(false);
+  mobilePopupHelper(menuOpened, setMenuOpened)
+
+  return (
+    <div style={{width: "100%"}}>
```
Comment (chiaberry) [inline src/components/PageSections/Pagination/Filter.js:46]: what is the default width?

**Review by chiaberry — APPROVED** [review]

File: src/js/i18n/definitions.js (hunk 3 above)
Comment (Niicck) [reply src/js/i18n/definitions.js:612]: you're right!

File: src/components/PageSections/Pagination/Filter.js (hunk 5 above)
Comment (Niicck) [reply src/components/PageSections/Pagination/Filter.js:46]: It looks like me adding "width:100%" makes no difference! It did at one point in development, but now everything has changed around it so our container always fills the full width of the screen.

**Pull request merged**